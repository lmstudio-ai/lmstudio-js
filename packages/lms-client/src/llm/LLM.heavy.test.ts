import { Chat, type LLM, LMStudioClient } from "../index.js";
import { ensureHeavyTestsEnvironment, llmTestingQwen05B } from "../shared.heavy.test.js";

describe("LLM", () => {
  let client: LMStudioClient;
  let model: LLM;
  const chat = Chat.from([
    { role: "system", content: "This is the system prompt." },
    { role: "user", content: "User message 1" },
    { role: "assistant", content: "Assistant message 1" },
    { role: "user", content: "User message 2" },
  ]);
  beforeAll(async () => {
    client = new LMStudioClient();
    await ensureHeavyTestsEnvironment(client);
  });
  beforeEach(async () => {
    model = await client.llm.model(llmTestingQwen05B, { verbose: false });
  });
  it("can apply prompt template to a regular chat", async () => {
    const formatted = await model.applyPromptTemplate(chat);
    expect(formatted).toMatchSnapshot();
  });
  it("can get model context length", async () => {
    const contextLength = await model.getContextLength();
    expect(contextLength).toMatchInlineSnapshot(`4096`);
  });
  it("can get model info", async () => {
    const modelInfo = await model.getModelInfo();
    expect(modelInfo).toMatchSnapshot({
      identifier: expect.any(String),
      instanceReference: expect.any(String),
      modelKey: expect.any(String),
    });
  });
  it("Can tokenize correctly", async () => {
    const tokens = await model.tokenize("Chaos is a ladder.");
    expect(tokens).toMatchSnapshot();
  });
  it("Can count tokens correctly", async () => {
    const count = await model.countTokens("Chaos is a ladder.");
    expect(count).toMatchInlineSnapshot(`6`);
  });
  it("Has correct properties", async () => {
    expect(model.displayName).toMatchInlineSnapshot(`"Qwen2.5 0.5B Instruct"`);
    expect(model.format).toMatchInlineSnapshot(`"gguf"`);
    expect(model.identifier).toEqual(llmTestingQwen05B);
    expect(model.path).toEqual(llmTestingQwen05B);
    expect(model.sizeBytes).toMatchInlineSnapshot(`397807936`);
    expect(model.trainedForToolUse).toMatchInlineSnapshot(`true`);
    expect(model.vision).toMatchInlineSnapshot(`false`);
  });
});
