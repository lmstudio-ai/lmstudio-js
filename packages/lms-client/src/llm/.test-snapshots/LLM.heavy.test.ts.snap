// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LLM Can tokenize correctly 1`] = `
[
  1143,
  64866,
  374,
  264,
  35765,
  13,
]
`;

exports[`LLM can apply prompt template to a regular chat 1`] = `
"<|im_start|>system
This is the system prompt.<|im_end|>
<|im_start|>user
User message 1<|im_end|>
<|im_start|>assistant
Assistant message 1<|im_end|>
<|im_start|>user
User message 2<|im_end|>
<|im_start|>assistant
"
`;

exports[`LLM can get model info 1`] = `
{
  "architecture": "qwen2",
  "contextLength": 4096,
  "displayName": "Qwen2.5 0.5B Instruct",
  "format": "gguf",
  "identifier": Any<String>,
  "instanceReference": Any<String>,
  "maxContextLength": 32768,
  "modelKey": Any<String>,
  "paramsString": "1B",
  "path": "lmstudio-community/Qwen2.5-0.5B-Instruct-GGUF/Qwen2.5-0.5B-Instruct-Q4_K_M.gguf",
  "sizeBytes": 397807936,
  "trainedForToolUse": true,
  "type": "llm",
  "vision": false,
}
`;
